arroyo-sql/src/tables.rs

```diff
@@ -646,10 +648,32 @@ impl Table {
                             .collect(),
                     }))
                 }
-                Some(connector) => Ok(Some(Table::ConnectorTable(
-                    ConnectorTable::from_options(&name, connector, fields, &mut with_map)
+                Some(connector) => {
+                    let connection_profile = match with_map.remove("connection_profile") {
```

Comment:
It doesn't appear that we validate that the profile is for the correct connector. I think that's worth doing so we can give a reasonable error to the user.

Reply:
Done